From b3764948c649619f3ccdc21b4a25b78133c8c09c Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 250/296] If child window is not obscuring screen, wait for
 main window drawn

When PopupWindow is showing, this child window may
be drawn earlier than it's main window, causing
starting-window to remove early.
---
 services/core/java/com/android/server/wm/AppWindowToken.java | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/services/core/java/com/android/server/wm/AppWindowToken.java b/services/core/java/com/android/server/wm/AppWindowToken.java
index 3f797344338..57186c2cda4 100644
--- a/services/core/java/com/android/server/wm/AppWindowToken.java
+++ b/services/core/java/com/android/server/wm/AppWindowToken.java
@@ -255,6 +255,11 @@ class AppWindowToken extends WindowToken implements WindowManagerService.AppFree
     }
 
     void onFirstWindowDrawn(WindowState win, WindowStateAnimator winAnimator) {
+        // Child window may cause removing starting window early
+        if (win.isChildWindow() && !win.fillsDisplay()) {
+            return;
+        }
+
         firstWindowDrawn = true;
 
         // We now have a good window to show, remove dead placeholders
-- 
2.11.0

